### PR TITLE
crude add embed service

### PR DESCRIPTION
**Pull request by Jerimee**
doesn't really work - having CSP problem

### CODE REVIEW

File: includes/EmbedService/EmbedServiceFactory.php [hunk 1]
```diff
@@ -86,6 +88,12 @@ public static function newFromName( string $serviceName, string $id ): AbstractE
 			case 'youtubevideolist':
 				return new YouTubeVideoList( $id );
 
+			case 'indiana':
+				return new Indiana( $id );
+
+			case 'itest':
+				return new Itest( $id );
+
```
Comment (octfx) [inline includes/EmbedService/EmbedServiceFactory.php:96]: This seems to be the same as Indiana

File: includes/EmbedService/Itest.php [hunk 2]
```diff
@@ -0,0 +1,66 @@
+<?php
+
+declare( strict_types=1 );
+
+namespace MediaWiki\Extension\EmbedVideo\EmbedService;
+
+final class Indiana extends AbstractEmbedService {
```
Comment (octfx) [inline includes/EmbedService/Itest.php:7]: I don't see any difference to the Indiana service

File: includes/EmbedService/Indiana.php [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+final class Indiana extends AbstractEmbedService {
+
+	/**
+	 * @inheritDoc
+	 */
+	public function getBaseUrl(): string {
+		//return '//player.vimeo.com/video/%1$s';
+		//return '//purl.dlib.indiana.edu/iudl/media/%1$s' . '/embed';
+		return '//media.dlib.indiana.edu/master_files/%1$s' . '/embed';
```
Comment (octfx) [inline includes/EmbedService/Indiana.php:15]: Could you add an example to the readme with a typical url for this service?

File: includes/EmbedService/Indiana.php [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+	protected function getDefaultHeight(): int {
+		return 450;
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	protected function getUrlRegex(): array {
+		return [
+			'Indiana'
```
Comment (octfx) [inline includes/EmbedService/Indiana.php:44]: This part is used to validate the URL of the service. I'm guessing something like the following could work:
`#media\.dlib\.indiana\.edu/master_files/[\d\w\-_]{3,}/embed#i`

To check if this works, you can paste the code into the input called `Regular Expression` at https://regex101.com and the url of a typical embed into the input called  `Test String` 

The test is successful, when the url is highlightet in blue

**Review by octfx — CHANGES_REQUESTED** [review]

**Comment by Jerimee** [comment]
Thank you for all these helpful comments. Very kind.

I'm having trouble getting past Indiana's CSP - perhaps this might be
something you can assist with?

I'll post a proper question later today

**Comment by octfx** [comment]
Sure thing!  
I'd need an example of an emeb to see which urls are needed

**Pull request closed without merging**